### PR TITLE
Add an argument to `swapteams` command

### PR DESCRIPTION
**Pull request by ShadowsAdi**
Related to: https://github.com/s1lentq/ReGameDLL_CS/issues/728

### CODE REVIEW

File: regamedll/dlls/game.cpp [hunk 1]
```diff
@@ -197,8 +197,17 @@ void GameDLL_EndRound_f()
...
+	{
+		const char *pCmd = CMD_ARGV(1);
+		
+		CSGameRules()->SwapAllPlayers();
+		
+		if (pCmd[0] == '1' || !Q_stricmp(pCmd, "true"))
+		{
+			CVAR_SET_FLOAT("sv_restartround", 1.0);
+		}
+	}
```
Comment (wopox1337) [inline regamedll/dlls/game.cpp:210]: now if you send swapteams without arguments, nothing will happen? Let's assume by default that the argument = 0

**Comment by wopox1337** [comment]
@ShadowsAdi 

1) You broke the default behavior of the `swapteams` command (without arguments).
2) You don't need to make boolean notation (true\false) for CVars. It was always 1\0.

File: regamedll/dlls/game.cpp [hunk 2]
```diff
@@ -197,8 +197,17 @@ void GameDLL_EndRound_f()
 
 void GameDLL_SwapTeams_f()
 {
-	CSGameRules()->SwapAllPlayers();
-	CVAR_SET_FLOAT("sv_restartround", 1.0);
+	if (CMD_ARGC() == 2)
```
Comment (wopox1337) [inline regamedll/dlls/game.cpp:200]: You broke the default behavior of the swapteams command (without arguments).

File: regamedll/dlls/game.cpp [hunk 3]
```diff
@@ -197,8 +197,17 @@ void GameDLL_EndRound_f()
...
+	{
+		const char *pCmd = CMD_ARGV(1);
+		
+		CSGameRules()->SwapAllPlayers();
+		
+		if (pCmd[0] == '1' || !Q_stricmp(pCmd, "true"))
```
Comment (wopox1337) [inline regamedll/dlls/game.cpp]: You don't need to make boolean notation (true\false) for CVars. It was always 1\0.

**Review by wopox1337 — APPROVED** [review]

**Comment by rust2** [comment]
What's the purpose of this change?

**Comment by wopox1337** [comment]
> What's the purpose of this change and how does it relate to #728?

Ability to execute the command `swapteams`, swap commands and control the restart of the game (as well as not to use the restart).


from your issue:
> Could be added some option to `swapteams` command that **will swap teams but also save team scores, frags and deaths**?

**Comment by rust2** [comment]
yes, very nice
so just swap teams but not force the round to restart???
then it was almost useless change
de_dust20000

**Pull request merged**